crates/flux-fhir-analysis/src/wf/sortck.rs

```diff
@@ -507,15 +512,36 @@ impl<'genv, 'tcx> InferCtxt<'genv, 'tcx> {
             .clone()
     }
 
+    fn normalize_projection_sort(
+        genv: GlobalEnv,
+        owner: FluxOwnerId,
+        sort: rty::Sort,
+    ) -> rty::Sort {
+        let infcx = genv
+            .tcx()
+            .infer_ctxt()
+            .build(TypingMode::non_body_analysis());
+        if let Some(def_id) = owner.def_id()
+            && let Ok(sort) = sort.normalize_sorts(def_id.into(), genv, &infcx)
+        {
+            sort
+        } else {
+            sort
+        }
+    }
+
     // FIXME(nilehmann) this is a bit of a hack. We should find a more robust way to do normalization
-    // for sort checking, including normalizing projections. Maybe we can do lazy normalization. Once
-    // we do that maybe we can also stop expanding aliases in `ConvCtxt::conv_sort`.
-    pub(crate) fn normalize_weak_alias_sorts(&mut self) -> QueryResult {
+    // for sort checking, including normalizing projections. [RJ: this is done now]
+    // Maybe we can do lazy normalization. Once we do that maybe we can also stop
+    // expanding aliases in `ConvCtxt::conv_sort`.
+    pub(crate) fn normalize_sorts(&mut self) -> QueryResult {
+        let genv = self.genv;
         for sort in self.sort_of_bty.values_mut() {
-            *sort = self.genv.deep_normalize_weak_alias_sorts(sort)?;
+            let sort1 = genv.deep_normalize_weak_alias_sorts(sort)?;
```

Comment:
We don't need to call `deep_normalize_weak_alias_sort` because `normalize_sorts` already normalizes weak aliases (I think)

Reply:
Good point, I removed the call.